torch/utils/_config_module.py

```diff
@@ -326,6 +321,22 @@ def __init__(self, config: _Config):
                     self.env_value_force = env_value
                     break
 
+        # Ensure justknobs and envvars are allowlisted types
+        if self.justknob is None:
+            if self.value_type is not None and (
+                config.env_name_default is not None or config.env_name_force is not None
+            ):
+                assert self.value_type in (
+                    bool,
+                    str,
+                    Optional[bool],
+                    Optional[str],
+                ), f"envvar configs only support (optional) booleans or strings, {self.value_type} is neither"
+        else:
+            assert isinstance(
+                self.default, bool
```

Comment:
nit - We should also allowed Optional[bool] here (Not blocking this PR).

Reply:
Added.